### ISSUE TITLE
Compiled module not found when using workers without module features

### ISSUE THREAD

**Issue by Soremwar** [denoland/deno#11060]
When running a worker for the first time (it apparently works after insisting for some time) Deno will throw an error indicating that the worker is not compiled and hang the process, requiring it to be killed manually. This is a dealbreaker when working with web workers and docker containers

```ts
// mod.ts
const worker = new Worker(new URL("./worker.ts", import.meta.url).href, {
  type: "module",
});

worker.onmessage = ({ data }) => {
  console.log(data);
};
worker.postMessage("important data");

export {};
```

```ts
// worker.ts
/// <reference no-default-lib="true" />
/// <reference lib="deno.worker" />

onmessage = ({ data }) => {
  postMessage(`"${data}" from worker`);
};
```

```log
deno run -A mod.ts
Check file:///deno_test/main.ts
Check file:///deno_test/worker.ts
warning: Compiled module not found "file:///deno_test/worker.ts"
  If the source module contains only types, use `import type` and `export type` to import it instead
```

An easy fix for this seems to be adding an `export {};` at the end of the worker file, but the fact that the error message is so cryptic will confuse a lot of users, like it happened to me

**Comment by kitsonk** [comment]
tsc us treating it as a script, but I don't know why it is refusing to emit.

**Comment by bartlomieju** [comment]
Ref https://github.com/denoland/deno/issues/9593

**Comment by bartlomieju** [comment]
I can confirm that this is still an issue, however now Deno panics:
```
============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos x86_64
Version: 1.17.3+b10563c
Args: ["deno", "run", "-A", "mod.ts"]

thread 'worker-0' panicked at 'internal error: entered unreachable code: Unexpected missing emit: file:///Users/biwanczuk/dev/deno/worker.ts', cli/proc_state.rs:587:25
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

**Comment by pokatomnik** [comment]
> This no longer occurs for me. Probably fixed by #14875.

Hello, could you please provide the code you compiled?
I am using Deno version 1.32.1 and this  is my code:
```typescript
// main.ts
const thread = new Worker(new URL("./worker.ts", import.meta.url).href, {
  type: "module",
});

thread.postMessage({ seconds: 10 });
thread.addEventListener("message", (evt) => {
  console.log(evt.data);
});

setInterval(() => {
  console.log("I am free!");
}, 1000);
```

```typescript
// worker.ts

/// <reference lib="webworker" />

function cpuIntencive(seconds: number) {
  const start = Date.now();
  const end = start + 1000 * seconds;
  let now = start;
  while (now <= end) {
    now = Date.now();
  }
}

self.onmessage = (e) => {
  const seconds: number = e.data.seconds;
  cpuIntencive(seconds);
  postMessage("ID_DONE");
};

```
I have successfully run this example with
```sh
$ deno run -A ./main.ts
```
But after compilation
```
deno compile -A --unstable ./main.ts
```
with and without `--unstable` flag I see this error:
>error: Uncaught (in worker "") Module not found
>error: Uncaught (in promise) Error: Unhandled error in child worker.
>    at Worker.#pollControl (ext:runtime/11_workers.js:157:19)

Could you please provide the precise version you use and your code (if you can of course!)
Thanks in advance!

**Comment by pokatomnik** [comment]
> If you use `deno compile` you must explicitly tell Deno which additional files to include like so: `deno compile -A --unstable --include worker.ts ./main.ts`. See https://deno.com/blog/v1.32#deno-compile-support-for-web-workers-and-dynamic-import

Oh, wow! Works great, thank you!
